auth/auth/auth.py

```diff
@@ -263,6 +336,48 @@ async def callback(request):
     return aiohttp.web.HTTPFound(next_page)
 
 
+@routes.post('/api/v1alpha/user/{user}/create')
+@rest_authenticated_developers_only
+async def create_user(request: web.Request, userdata):  # pylint: disable=unused-argument
+    db: Database = request.app['db']
+    username = request.match_info['user']
+
+    body = await request.json()
+    login_id = body['login_id']
+    is_developer = body['is_developer']
+    is_service_account = body['is_service_account']
+
+    try:
+        await check_valid_new_user(db, username, login_id, is_developer, is_service_account)
+    except AuthUserError as e:
+        raise e.http_response()
+
+    await db.execute_insertone(
+        '''
+INSERT INTO users (state, username, login_id, is_developer, is_service_account)
+VALUES (%s, %s, %s, %s);
+''',
+        ('creating', username, login_id, is_developer, is_service_account),
+    )
+
+    async def _create():
+        delay = 5
+        while True:
+            user = await user_from_login_id(db, login_id)
+            if user['state'] != 'creating':
+                break
+            delay = await sleep_and_backoff(delay, max_delay=30)
+        return user
+
+    try:
+        user = await asyncio.wait_for(_create(), 300)
+    except asyncio.TimeoutError:
+        raise web.HTTPInternalServerError(body=f"timeout creating user '{username}' with login_id '{login_id}'")
```

Comment:
As a general design principle, I think HTTP endpoints should return immediately. We should add a separate endpoint `GET /api/v1alpha/users/{user}` which returns the user dict to a developer. The developer can poll that if waiting for creation is necessary.